sotodlib/site_pipeline/preprocess_tod.py

```diff
@@ -468,6 +469,53 @@ def load_preprocess_det_select(obs_id, configs, context=None,
     pipe[-1].select(meta)
     return meta
 
+def load_preprocess_tod_sim(obs_id, sim_map,
+                            configs="preprocess_configs.yaml",
+                            context=None, dets=None,
+                            meta=None, modulated=True):
+    """ Loads the saved information from the preprocessing pipeline and runs the
+    processing section of the pipeline on simulated data
+
+    Assumes preprocess_tod has already been run on the requested observation. 
+    
+    Arguments
+    ----------
+    obs_id: multiple
+        passed to `context.get_obs` to load AxisManager, see Notes for 
+        `context.get_obs`
+    sim_map: pixell.enmap.ndmap
+        signal map containing (T, Q, U) fields
+    configs: string or dictionary
+        config file or loaded config directory
+    dets: dict
+        dets to restrict on from info in det_info. See context.get_meta.
+    meta: AxisManager
+        Contains supporting metadata to use for loading.
+        Can be pre-restricted in any way. See context.get_meta.
+    modulated: bool
+        If True, apply the HWP angle model and scan the simulation
+        into a modulated signal.
+        If False, scan the simulation into demodulated timestreams.
+    """
+    configs, context = _get_preprocess_context(configs, context)
+    meta = load_preprocess_det_select(obs_id, configs=configs, context=context, dets=dets, meta=meta)
+
+    if meta.dets.count == 0:
+        logger.info(f"No detectors left after cuts in obs {obs_id}")
+        return None
+    else:
+        pipe = Pipeline(configs["process_pipe"], logger=logger)
+        aman = context.get_obs(meta, no_signal=True)
+        if modulated:
+            # Apply the HWP angle model here
+            # WARNING : should be turned off in the config file
+            # to filter simulations
+            aman = hwp_angle_model.apply_hwp_angle_model(aman)
+            aman.move("signal", None)
```

Comment:
But you ran `get_obs` with `no_signal = True` so why do you have to delete the `signal` field?

Reply:
For some reason, the property signal already exists and I have to remove it before loading a simulation.
It is a quick fix to this issue